Automattic/pocket-casts-ios
# Search Improvements: create a search results delegate

| 📘 Project: #709  |
|:---:|

This PR adds a delegate so both the new and old search can be easily connected/disconnected from the app.

## To test

1. Run the app
1. In Profile > Settings > Beta features > make sure `newSearch` is disabled
1. ✅  Perform a search from both "Podcasts" and "Discover" and make sure it works as always
1. Go to Profile > Settings > Beta features > enable `newSearch`
1. ✅  When you tap the search, the search history should display
1. Tap any text into the search field
1. ✅ Search history should disappear and "Search results" should appear
1. Clear the text
1. ✅ Search history should appear again

## Checklist
